lib/src/database/matrix_sdk_database.dart

```diff
@@ -433,7 +434,7 @@ class MatrixSdkDatabase extends DatabaseApi {
                 ? timelineEventIds.getRange(start, end).toList()
                 : []);
 
-        return await _getEventsByIds(eventIds.cast<String>(), room);
+        return await _getEventsByIds(eventIds, room);
```

Comment:
This has already been fixed here: https://github.com/famedly/matrix-dart-sdk/commit/2afd7040a7fa7f8a65cab6196c0f56407334dc65